BraintreeModels/src/main/java/com/braintreepayments/api/PaymentMethodNonce.java

```diff
@@ -176,11 +80,42 @@ public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(mNonce);
         dest.writeString(mDescription);
         dest.writeByte(mDefault ? (byte) 1 : (byte) 0);
+        dest.writeInt(mType);
     }
 
     protected PaymentMethodNonce(Parcel in) {
         mNonce = in.readString();
         mDescription = in.readString();
         mDefault = in.readByte() > 0;
+        mType = in.readInt();
+    }
+
+    public static final Creator<PaymentMethodNonce> CREATOR = new Creator<PaymentMethodNonce>() {
+        @Override
+        public PaymentMethodNonce createFromParcel(Parcel in) {
+            return new PaymentMethodNonce(in);
+        }
+
+        @Override
+        public PaymentMethodNonce[] newArray(int size) {
+            return new PaymentMethodNonce[size];
+        }
+    };
+
+    static @PaymentMethodType int paymentMethodTypeFromString(String typeString) {
```

Comment:
Why not have this in the PaymentMethodType.java file?

Reply:
+1 to this idea. Could we even use `StringDef` instead of `IntDef` so that the enum is backed by the associated string instead of an int?

Reply:
Yeah, I agree we should move it. The only concern with `StringDef` is it doesn't account for `PaymentMethodType.UNKNOWN`, so we'd still need to have some switching logic to convert from `String` to `PaymentMethodType`.

Reply:
We tried to move it and I forgot that the compiler barked when we tried to add a static method to an interface. We may just make the method `private` for now and keep it in PaymentMethodNonce.java until we can find a more suitable solution.